src/NHibernate/Linq/GroupBy/AggregatingGroupByRewriter.cs

```diff
@@ -104,5 +104,24 @@ private static void FlattenSubQuery(QueryModel queryModel, QueryModel subQueryMo
 			// Replace the outer query source
 			queryModel.MainFromClause = subQueryModel.MainFromClause;
 		}
+
+		private static IEnumerable<Expression> RemoveCostantGroupByKeys(QueryModel queryModel, GroupResultOperator groupBy)
+		{
+			var keys = groupBy.ExtractKeyExpressions().Where(x => !(x is ConstantExpression)).ToList();
+
+			if (!keys.Any())
+			{
+				// Remove the Group By clause completely if all the keys are constant (redundant)
+				queryModel.ResultOperators.Remove(groupBy);
```

Comment:
Please include test for [NH-2500](https://nhibernate.jira.com/browse/NH-2500) (overcaching). It might be the case.